### ISSUE TITLE
ERR_INVALID_HTTP_TOKEN - h3 events

### ISSUE THREAD

**Issue by birkskyum** [nksaraf/vinxi#155]
don't know if this belong in vinxi or soild-start

I upgraded to latest vinxi (0.1.7) /solid-start (0.4.11), and now I get a lot of this when i start my dev server and go to a page:

```sh
> dev
> vinxi dev

vinxi 0.1.7
vinxi Found vite.config.js with app config
vinxi starting dev server

  ➜ Local:    http://localhost:3000/
  ➜ Network:  use --host to expose

[h3] [unhandled] TypeError [ERR_INVALID_HTTP_TOKEN]: Header name must be a valid HTTP token ["{
  request: [Request],
  clientAddress: '::1',
  locals: {},
  [Symbol(h3Event)]: [H3Event]
}"]
    at ServerResponse.setHeader (node:_http_outgoing:705:3)
    ... 12 lines matching cause stack trace ...
    at async Server.toNodeHandle (file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:1907:7) {
  cause: TypeError [ERR_INVALID_HTTP_TOKEN]: Header name must be a valid HTTP token ["{
    request: [Request],
    clientAddress: '::1',
    locals: {},
    [Symbol(h3Event)]: [H3Event]
  }"]
      at ServerResponse.setHeader (node:_http_outgoing:705:3)
      at setResponseHeader (file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:747:18)
      at Module.eval (/Users/admin/repos/project/docs/node_modules/vinxi/runtime/server.js:146:12)
      at Module.createPageEvent (/Users/admin/repos/project/docs/node_modules/@solidjs/start/server/page-event.tsx:27:25)
      at eval (/Users/admin/repos/project/docs/node_modules/@solidjs/start/entry/index.ts:35:53)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at Proxy.provideRequestEvent (file:///Users/admin/repos/project/docs/node_modules/solid-js/web/dist/storage.js:7:14)
      at handler (/Users/admin/repos/project/docs/node_modules/@solidjs/start/entry/index.ts:23:36)
      at _callHandler (file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:1563:22)
      at _handler (file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:1547:12)
      at Object.handler (/Users/admin/repos/project/docs/node_modules/vinxi/lib/router-modes.js:268:18)
      at async file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:1697:19
      at async Object.callAsync (/Users/admin/repos/project/docs/node_modules/unctx/dist/index.cjs:74:16)
      at async Server.toNodeHandle (file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:1907:7) {
    code: 'ERR_INVALID_HTTP_TOKEN'
  },
  statusCode: 500,
  fatal: false,
  unhandled: true,
  statusMessage: undefined,
  data: undefined
}
```

I can work on a repro if needed.

**Comment by nksaraf** [comment]
Oops this is a impendence mismatch between when we made a certain change and solid-start was able to reflect that change. Let me fix this quickly on our end and release it.

**Comment by nksaraf** [comment]
try v0.1.8, this should be fixed

**Comment by birkskyum** [comment]
I actually still see something similar:

```sh
vinxi 0.1.8
vinxi Found vite.config.js with app config
vinxi starting dev server

  ➜ Local:    http://localhost:3000/
  ➜ Network:  use --host to expose

[h3] [unhandled] TypeError [ERR_INVALID_HTTP_TOKEN]: Header name must be a valid HTTP token ["{
  request: [Request],
  clientAddress: '::1',
  locals: {},
  [Symbol(h3Event)]: [H3Event]
}"]
    at ServerResponse.setHeader (node:_http_outgoing:705:3)
    ... 12 lines matching cause stack trace ...
    at async Server.toNodeHandle (file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:1907:7) {
  cause: TypeError [ERR_INVALID_HTTP_TOKEN]: Header name must be a valid HTTP token ["{
    request: [Request],
    clientAddress: '::1',
    locals: {},
    [Symbol(h3Event)]: [H3Event]
  }"]
      at ServerResponse.setHeader (node:_http_outgoing:705:3)
      at setResponseHeader (file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:747:18)
      at Module.eval (/Users/admin/repos/project/docs/node_modules/vinxi/runtime/server.js:147:12)
      at Module.createPageEvent (/Users/admin/repos/project/docs/node_modules/@solidjs/start/server/page-event.tsx:27:25)
      at eval (/Users/admin/repos/project/docs/node_modules/@solidjs/start/entry/index.ts:35:53)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at Proxy.provideRequestEvent (file:///Users/admin/repos/project/docs/node_modules/solid-js/web/dist/storage.js:7:14)
      at handler (/Users/admin/repos/project/docs/node_modules/@solidjs/start/entry/index.ts:23:36)
      at _callHandler (file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:1563:22)
      at _handler (file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:1547:12)
      at Object.handler (/Users/admin/repos/project/docs/node_modules/vinxi/lib/router-modes.js:268:18)
      at async file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:1697:19
      at async Object.callAsync (/Users/admin/repos/project/docs/node_modules/unctx/dist/index.cjs:74:16)
      at async Server.toNodeHandle (file:///Users/admin/repos/project/docs/node_modules/h3/dist/index.mjs:1907:7) {
    code: 'ERR_INVALID_HTTP_TOKEN'
  },
  statusCode: 500,
  fatal: false,
  unhandled: true,
  statusMessage: undefined,
  data: undefined
}
```

**Comment by nksaraf** [comment]
sorry i meant 0.1.9

**Comment by nksaraf** [comment]
wait no should have been fixed in v0.1.8, can I see a repro by any chance?

**Comment by nksaraf** [comment]
I believe this should be fixed in vinxi 0.1.10

**Comment by birkskyum** [comment]
That did the trick! Thanks!
